test/common/cro_cr_success.go

```diff
@@ -49,67 +47,63 @@ func TestCROPostgresSuccessfulState(t *testing.T, ctx *TestingContext) {
 		}
 
 		postgres := &crov1.Postgres{}
-		err := getResourceAndUnMarshalJsonToResource(ctx, "postgres", postgresName, postgres, testErrors)
+		err := getResourceAndUnMarshalJsonToResource(ctx, "postgres", postgresName, postgres)
 
-		if err == nil && postgres.Status.Phase != croTypes.PhaseComplete && postgres.Status.Strategy != strategy {
-			testErrors = append(testErrors, fmt.Sprintf("\n%s Postgres not ready with phase: %s, message: %s, provider, %s", postgresName, postgres.Status.Phase, postgres.Status.Message, postgres.Status.Provider))
+		if err != nil {
+			t.Errorf("Failed to retrieve postgres custom resource: %s", err)
 		}
-	}
 
-	if len(testErrors) != 0 {
-		t.Fatalf("Test CRO Postgress Succesful failed with the following errors: %s", testErrors)
+		if err == nil && postgres.Status.Phase != croTypes.PhaseComplete && postgres.Status.Strategy != strategy {
+			t.Errorf("%s Postgres not ready with phase: %s, message: %s, provider, %s", postgresName, postgres.Status.Phase, postgres.Status.Message, postgres.Status.Provider)
+		}
 	}
 }
 
 func TestCRORedisSuccessfulState(t *testing.T, ctx *TestingContext) {
-	var testErrors []string
 	strategy := getResourceStrategy(t, ctx)
 
 	for _, redisName := range redisToCheck {
 		redis := &crov1.Redis{}
-		err := getResourceAndUnMarshalJsonToResource(ctx, "redis", redisName, redis, testErrors)
+		err := getResourceAndUnMarshalJsonToResource(ctx, "redis", redisName, redis)
 
-		if err == nil && redis.Status.Phase != croTypes.PhaseComplete && redis.Status.Strategy != strategy {
-			testErrors = append(testErrors, fmt.Sprintf("\n%s redis not ready with phase: %s, message: %s, provider, %s", redisName, redis.Status.Phase, redis.Status.Message, redis.Status.Provider))
+		if err != nil {
+			t.Errorf("Failed to retrieve redis custom resource: %s", err)
 		}
-	}
 
-	if len(testErrors) != 0 {
-		t.Fatalf("Test CRO Redis Succesful failed with the following errors: %s", testErrors)
+		if err == nil && redis.Status.Phase != croTypes.PhaseComplete && redis.Status.Strategy != strategy {
```

Comment:
I think is nicer to use a `continue` in the previous `if` than checking again if the error is `nil`, wdyt?

Example:
https://github.com/integr8ly/integreatly-operator/blob/69161d395707aefabacf7dd12e1e851d672d61be/test/common/deployment_types.go#L279

Reply:
Sounds good to me, I'll make that change now :+1:

Reply:
Good suggestion ! I'll make this change now :+1: